pkg/api/unversioned/time.go

```diff
@@ -79,6 +80,16 @@ func (t Time) Equal(u Time) bool {
 	return t.Time.Equal(u.Time)
 }
 
+// BeforeMicro reports whether the time instant t is before micro instance u.
+func (t Time) BeforeMicro(u v1.MicroTime) bool {
+	return t.Time.Before(u.Time)
```

Comment:
This shouldn't be needed - both Time and MicroTime, as go-structs have full precision. The only difference is that when serialized to JSON the latter is more precise.